quickstarts/microsoft.appconfiguration/app-configuration/main.bicep

```diff
@@ -0,0 +1,161 @@
+
+@description('Specify the storage account name.')
+param storageAccountName string
+
+@description('Name of the resource group for the app config store.')
+param appConfigStoreResourceGroup string
+
+@description('App configuration store name.')
+param appConfigStoreName string
+
+@description('Specify the name of the key in the app config store for the VM windows sku.')
+param vmSkuKey string
+
+@description('Specify the name of the key in the app config store for the VM disk size')
+param diskSizeKey string
+
+var nicName = 'myVMNic'
+var addressPrefix = '10.0.0.0/16'
+var subnetName = 'Subnet'
+var subnetPrefix = '10.0.0.0/24'
+var publicIPAddressName = 'myPublicIP'
+var vmName = 'SimpleWinVM'
+var virtualNetworkName = 'MyVNET'
+var subnetRef = resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetworkName, subnetName)
+var appConfigRef = resourceId(appConfigStoreResourceGroup, 'Microsoft.AppConfiguration/configurationStores', appConfigStoreName)
+var windowsOSVersionParameters = {
+  key: vmSkuKey
+  label: 'template'
+}
+var diskSizeGBParameters = {
+  key: diskSizeKey
+  label: 'template'
+}
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2021-08-01' = {
+  name: storageAccountName
+  location: location
+  sku: {
+    name: 'Standard_LRS'
+  }
+  kind: 'Storage'
+  properties: {}
+}
+
+resource publicIPAddress 'Microsoft.Network/publicIPAddresses@2021-05-01' = {
+  name: publicIPAddressName
+  location: location
+  properties: {
+    publicIPAllocationMethod: 'Dynamic'
+    dnsSettings: {
+      domainNameLabel: domainNameLabel
+    }
+  }
+}
+
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2021-05-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        addressPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: subnetPrefix
+        }
+      }
+    ]
+  }
+}
+
+resource nic 'Microsoft.Network/networkInterfaces@2021-05-01' = {
+  name: nicName
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig1'
+        properties: {
+          privateIPAllocationMethod: 'Dynamic'
+          publicIPAddress: {
+            id: publicIPAddress.id
+          }
+          subnet: {
+            id: subnetRef
+          }
+        }
+      }
+    ]
+  }
+  dependsOn: [
+    virtualNetwork
+  ]
+}
+
+resource vm 'Microsoft.Compute/virtualMachines@2021-11-01' = {
+  name: vmName
+  location: location
+  properties: {
+    hardwareProfile: {
+      vmSize: vmSize
+    }
+    osProfile: {
+      computerName: vmName
+      adminUsername: adminUsername
+      adminPassword: adminPassword
+    }
+    storageProfile: {
+      imageReference: {
+        publisher: 'MicrosoftWindowsServer'
+        offer: 'WindowsServer'
+        sku: listKeyValue(appConfigRef, '2019-10-01', windowsOSVersionParameters).value
```

Comment:
use the symbolic reference for getting the keys (here and below)

Reply:
Resolved in latest commit